core/templates/dev/head/services/ContextService.js

```diff
@@ -100,8 +109,29 @@ oppia.factory('ExplorationContextService', [
           }
 
           throw Error(
-            'ERROR: ExplorationContextService should not be used outside the ' +
-            'context of an exploration.');
+            'ERROR: ContextService should not be used outside the ' +
+            'context of an exploration or a question.');
+        }
+      },
+
+      // Returns a string representing the questionId (obtained from the
+      // URL).
+      getQuestionId: function() {
+        if (questionId) {
+          return questionId;
+        } else {
+          // The pathname should /question_editor/{question_id}.
+          var pathnameArray = UrlService.getPathname().split('/');
+          for (var i = 0; i < pathnameArray.length; i++) {
+            if (pathnameArray[i] === 'question_editor') {
+              questionId = pathnameArray[i + 1];
+              return pathnameArray[i + 1];
+            }
+          }
+
+          throw Error(
+            'ERROR: ContextService should not be used outside the ' +
+            'context of an question or a question.');
```

Comment:
Error message doesn't make sense.

Reply:
This error is thrown when the service is used outside the context of question and exploration.

Reply:
How about "ERROR: ContextService should be used in context of either an Exploration or a Question."